data/txHandler.go

```diff
@@ -220,31 +222,72 @@ func (handler *TxHandler) asyncVerifySignature(arg interface{}) interface{} {
 	return nil
 }
 
+// attempt to preallocate buffer on stack:
+// hardcode maxTxGroupSize constant in order to get rid ledger lookups and use backing arrays,
+// or fallback to heap allocation it MaxTxGroupSize is greater than our constant
+const maxTxGroupSize = 16
+
 func (handler *TxHandler) processIncomingTxn(rawmsg network.IncomingMessage) network.OutgoingMessage {
+	var decTxGroupElemBackingArray [maxTxGroupSize]int
+	var decTxGroupElemOffsets []int // offsets of txn ends
+
+	if maxTxGroupSize < config.MaxTxGroupSize {
+		decTxGroupElemOffsets = make([]int, config.MaxTxGroupSize)
+	} else {
+		decTxGroupElemOffsets = decTxGroupElemBackingArray[:]
+	}
+
+	unverifiedTxGroup := make([]transactions.SignedTxn, 1)
```

Comment:
I tried different tricks to have `unverifiedTxGroup` inside a stack-based backing array but go escapes it to heap probably because of referencing its elements in `Decode`